src/compiler/checker.ts

```diff
@@ -161,6 +161,8 @@ namespace ts {
 
         let jsxElementClassType: Type;
 
+        let deferredNodes: Node[];

```

Comment:
`deferredCheckNodes` and leave a decent comment about why these need to be deferred until the end.